oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/DefaultReactiveOAuth2AuthorizedClientManager.java

```diff
@@ -190,4 +338,128 @@ public void setContextAttributesMapper(Function<OAuth2AuthorizeRequest, Mono<Map
 					.defaultIfEmpty(Collections.emptyMap());
 		}
 	}
+
+	/**
+	 * The default authorization success handler,
+	 * which will save the authorized client in a
+	 * {@link ServerOAuth2AuthorizedClientRepository}.
+	 *
+	 * @since 5.3
+	 */
+	public static class DefaultAuthorizationSuccessHandler implements ReactiveOAuth2AuthorizationSuccessHandler {
+
+		/**
+		 * The repository in which authorized clients will be saved.
+		 */
+		private final ServerOAuth2AuthorizedClientRepository authorizedClientRepository;
+
+		/**
+		 * @param authorizedClientRepository The repository in which authorized clients will be saved.
+		 */
+		public DefaultAuthorizationSuccessHandler(ServerOAuth2AuthorizedClientRepository authorizedClientRepository) {
+			Assert.notNull(authorizedClientRepository, "authorizedClientRepository cannot be null");
+			this.authorizedClientRepository = authorizedClientRepository;
+		}
+
+		@Override
+		public Mono<Void> onAuthorizationSuccess(OAuth2AuthorizationSuccessEvent authorizationSuccessEvent) {
+			return this.authorizedClientRepository.saveAuthorizedClient(
+					authorizationSuccessEvent.getAuthorizedClient(),
+					authorizationSuccessEvent.getPrincipal(),
+					authorizationSuccessEvent.getAttribute(ServerWebExchange.class.getName()));
+		}
+	}
+
+	/**
+	 * The default authorization failure handler,
+	 * which will remove the authorized client from a
+	 * {@link ServerOAuth2AuthorizedClientRepository}
+	 * for specific OAuth 2.0 error codes.
+	 *
+	 * @since 5.3
+	 */
+	public static class DefaultAuthorizationFailureHandler implements ReactiveOAuth2AuthorizationFailureHandler {
```

Comment:
Same comments apply from `DefaultAuthorizationSuccessHandler `

Reply:
Pulled out into public outer class as per other discussion in ServerOAuth2AuthorizedClientExchangeFilterFunction